### PR TITLE
DEX-161 Clear agendas

### PR DESCRIPTION
**Pull request by harzo**

### CODE REVIEW

**Comment by jksf** [comment]
I cannot start `exccd`. I get:
```
2018-06-20 17:13:55.161 [ERR] EXCC: Unable to start server on [:9666]: deployment ID sdiffalgorithm does not exist
```

**Comment by harzo** [comment]
Yes, it seems that agendas which had an impact on implementation needs to stay in params.go.
Already reverted SDiff agenda.

**Comment by jksf** [comment]
Is it just me or you too get looped error:
```
2018-06-22 13:04:59.172 [ERR] MINR: Failed to create new block template: deployment ID lnfeatures does not exist
```

**Comment by harzo** [comment]
Hmm i didn't notice this yesterday but now i get it too. 
Maybe clearing the agendas was a mistake.
Let me just change Expiry Date for LN to the past so it won't appearing at stakepools `Voting` page.

**Comment by harzo** [comment]
@jksf Regarding `difficulty tests` which are skipped - each change in Stake/Subsidy parameters and in Premine alters actual values in difficulty tests, so we need to fix it each time if we want the tests to pass.

I have confirmed this by reversing all Parameters and Premine values to the origin ones and tests were passing.

File: blockchain/validate.go [hunk 1]
```diff
@@ -1131,19 +1131,12 @@ func (b *BlockChain) checkBlockContext(block *exccutil.Block, prevNode *blockNod
 		// Switch to using the past median time of the block prior to
 		// the block being checked for all checks related to lock times
 		// once the stake vote for the agenda is active.
-		blockTime := header.Timestamp
-		lnFeaturesActive, err := b.isLNFeaturesAgendaActive(prevNode)
+		medianTime, err := b.index.CalcPastMedianTime(prevNode)
```
Comment (jksf) [inline blockchain/validate.go]: Can we just assign it to `blockTime`? `medianTime` is not used anymore.

Comment (harzo) [reply blockchain/validate.go]: Sure, good point :+1:

File: server.go [hunk 2]
```diff
@@ -2482,15 +2482,9 @@ out:
...
-	// active.
-	isActive, err := chain.IsLNFeaturesAgendaActive()
-	if err != nil {
-		return 0, err
-	}
-	if isActive {
-		scriptFlags |= txscript.ScriptVerifySHA256
-	}
+	// Enable validation of OP_SHA256 for the LN agenda
+	scriptFlags |= txscript.ScriptVerifySHA256
```
Comment (jksf) [inline server.go]: Maybe we should add `ScriptVerifySHA256` to `BaseStandardVerifyFlags`?

File: chaincfg/premine.go [hunk 3]
```diff
@@ -3190,11 +3190,15 @@ var BlockOneLedgerMainNet = []*TokenPayout{
 
 // BlockOneLedgerTestNet is the block one output ledger for the test
 // network.
-var BlockOneLedgerTestNet = []*TokenPayout{}
+var BlockOneLedgerTestNet = []*TokenPayout{
```
Comment (jksf) [inline chaincfg/premine.go]: `BlockOneLedgerTestNet` is not used at all

File: chaincfg/premine.go [hunk 4]
```diff
@@ -3190,11 +3190,15 @@ var BlockOneLedgerMainNet = []*TokenPayout{
...
+	{"TsmWaPM77WSyA3aiQ2Q1KnwGDVWvEkhipBc", 100000 * 1e8},
+}
 
 // BlockOneLedgerTestNet2 is the block one output ledger for the 2nd test
 // network.
-var BlockOneLedgerTestNet2 = []*TokenPayout{}
+var BlockOneLedgerTestNet2 = []*TokenPayout{
+	{"TsmWaPM77WSyA3aiQ2Q1KnwGDVWvEkhipBc", 100000 * 1e8},
```
Comment (jksf) [inline chaincfg/premine.go]: Do we have private keys or mnemonic for this address?

**Review by jksf — APPROVED** [review]

**Pull request merged**